pkg/server/download.go

```diff
@@ -355,3 +357,24 @@ func (s *server) updateDownloadSubscribers(total int64, current int64, name stri
 		s.subscribers.Delete(id)
 	}
 }
+
+// If the source and destination are on different file systems this will cause
+// an invalid cross-link device error using os.Rename. Instead, open the file
+// and then write it out
+func fileSystemSafeRename(tempPath string, targetLocation string) error {
+	byteData, err := os.ReadFile(tempPath)
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(targetLocation, byteData, 0644)
+	if err != nil {
+		return err
+	}
+	err = os.Remove(tempPath)
+
+	// Copy successful, dont error on temp file delete failure
+	if err != nil {
+		log.Error(fmt.Sprintf("Failed to remove temp file: %v", err))
```

Comment:
might be helpful to add the temp file name to the error message in case the error doesn't naturally contain it
```
log.Error(fmt.Errorf("Failed to remove temp file %s: %w", tempPath, err))
```